org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/model/ScalaThread.scala

```diff
@@ -55,12 +53,20 @@ abstract class ScalaThread private (target: ScalaDebugTarget, val threadRef: Thr
   override def isStepping: Boolean = ???
   private def canStep = suspended && !target.isPerformingHotCodeReplace
 
-  override def stepInto(): Unit = stepIntoFrame(stackFrames.head)
+  override def stepInto(): Unit = {
+    for (head <- stackFrames.headOption) {
+      wrapJDIException("Exception while performing `step into`") { ScalaStepInto(head).step() }
+    }
+  }
```

Comment:
i would use `foreach` literally because option here doesn't mimic a seq here, but it's my opinion